modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/commerce_marketplace_payment_method/MarketplaceViews.tsx

```diff
@@ -78,6 +66,19 @@ export default function MarketplaceViews() {
 		[placedOrders]
 	);
 
+	const getButtonConfiguration = (product: Product) => {
+		return isProductInstalled(product)
+			? {
+					disabled: true,
+					title: Liferay.Language.get('installed'),
+				}
+			: {
+					disabled: !new MarketplaceProduct(
+						product
+					).hasPermissionToInstall(permissions),
+				};
```

Comment:
For better understanding

```suggestion
                if (isProductInstalled(product)) {
                    return {
                        disabled: true,
			title: Liferay.Language.get('installed'),
                    }
                }
                
               const marketplaceProduct = new MarketplaceProduct(product);
                
                if (!marketplaceProduct.hasPermissionToInstall(permissions)) {
                        return {
                            disabled: true,
			    title: Liferay.Language.get('no-permissions'),
                        }
                }
```